File: python/tekst-zad2/69.1.py
fp = open('dane_gen.txt', 'r')
arr = fp.read().splitlines()
fp.close()
#-----------------------------------------------------------

tempArr = [0]*501

for subject in arr:
    tempArr[len(subject)] += 1

species = 0
for num in tempArr:
    if num != 0:
        species += 1

print(f"Liczba gatunkow: {species}")
print(f"Najliczniejszy gatunek: {max(tempArr)}")

File: c++/tekst-zad2/69/69.1.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <algorithm>

using namespace std;

int main(){
    ifstream fp;
    fp.open("dane_gen.txt");
    if(!fp.good()){
        return 0;
    }
    string arr[1000];
    for(int i = 0; i<1000; i++){
        fp >> arr[i];
    }
    fp.close();
    int lengths[501] = {0};
    for(int i = 0; i<1000; i++){
        lengths[arr[i].length()]++;
    }
    int species = 0;
    for(int i = 0; i<501; i++){
        if(lengths[i] > 0){
            species++;
        }
    }
    cout << species << '\n';
    cout << *max_element(lengths, lengths+500) << '\n';

    return 0;
}

//by Jakub Rekas